src/Tasks/GenerateResource.cs

```diff
@@ -1695,58 +1724,38 @@ private void GetStronglyTypedResourceToProcess(ref List<ITaskItem> inputsToProce
         /// Returns the newest last write time among the references and additional inputs.
         /// If any do not exist, returns DateTime.MaxValue so that resource generation produces a nice error.
         /// </summary>
-        private DateTime GetNewestUncorrelatedInputWriteTime()
+        private void UpdateNewestUncorrelatedInputWriteTime()
         {
-            if (!_foundNewestUncorrelatedInputWriteTime)
+            if (_newestUncorrelatedInput != null)
             {
-                _newestUncorrelatedInputWriteTime = DateTime.MinValue;
+                // We already did this
+                return;
+            }
 
-                // Check the timestamp of each of the passed-in references to find the newest
-                if (this.References != null)
-                {
-                    foreach (ITaskItem reference in this.References)
-                    {
-                        DateTime referenceTime = NativeMethodsShared.GetLastWriteFileUtcTime(reference.ItemSpec);
+            // Check the timestamp of each of the passed-in references to find the newest;
+            // and then the additional inputs
+            var inputs = (this.References ?? Enumerable.Empty<ITaskItem>()).Concat(this.AdditionalInputs ?? Enumerable.Empty<ITaskItem>());
 
-                        if (referenceTime == DateTime.MinValue)
-                        {
-                            // File does not exist: force a build to produce an error message
-                            _foundNewestUncorrelatedInputWriteTime = true;
-                            return DateTime.MaxValue;
```

Comment:
Note this method was previously returning MaxValue to force a build if any input was missing.

For clarity I changed to consistently using MinValue to mean a file is missing (same as GetLastWriteFileUtcTime returns for a missing file) and explicitly checking for MinValue as needed.